huaweicloud/resource_huaweicloud_networking_secgroup.go

```diff
@@ -124,47 +126,104 @@ func ResourceNetworkingSecGroup() *schema.Resource {
-	meta interface{}) diag.Diagnostics {
+func createSecurityGroupByV1Client(d *schema.ResourceData, config *config.Config) error {
+	region := GetRegion(d, config)
+	// The v3 API does not exist or has not been published in this region, retry creation using v1 client.
+	v1Client, err := config.SecurityGroupV1Client(region)
+	if err != nil {
+		return fmtp.Errorf("Error creating HuaweiCloud networking v1 client: %s", err)
+	}
+	createOpts := v1Groups.CreateOpts{
+		Name:                d.Get("name").(string),
+		EnterpriseProjectId: GetEnterpriseProjectID(d, config),
+	}
+	securityGroup, err := v1Groups.Create(v1Client, createOpts).Extract()
+	if err != nil {
+		return fmtp.Errorf("Error creating Security Group: %s", err)
+	}
+	d.SetId(securityGroup.ID)
+
+	if _, ok := d.GetOk("description"); ok {
+		v2Client, err := config.NetworkingV2Client(region)
+		if err != nil {
+			return fmtp.Errorf("Error creating HuaweiCloud networking v2 client: %s", err)
+		}
+
+		// The v1 API does not support creating and updating methods for description parameters.
+		err = updateSecurityGroupByV2Client(d, v2Client)
+		if err != nil {
+			return fmtp.Errorf("Error updating description of Security group (%s): %s", d.Id(), err)
+		}
+	}
+
+	// Delete the default security group rules if it has been requested.
+	if d.Get("delete_default_rules").(bool) {
+		for _, rule := range securityGroup.SecurityGroupRules {
+			if err := v1Rules.Delete(v1Client, rule.ID).ExtractErr(); err != nil {
+				return fmtp.Errorf("There was a problem deleting a default security group rule: %s", err)
+			}
+		}
+	}
+	return nil
+}
+
+func updateSecurityGroupByV2Client(d *schema.ResourceData, client *golangsdk.ServiceClient) error {
+	desc := d.Get("description").(string)
+	updateOpts := v2Groups.UpdateOpts{
+		Name:        d.Get("name").(string),
+		Description: &desc,
+	}
+	_, err := v2Groups.Update(client, d.Id(), updateOpts).Extract()
+	return err
+}
+
+func resourceNetworkingSecGroupCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
 	config := meta.(*config.Config)
-	client, err := config.NetworkingV3Client(GetRegion(d, config))
+	region := GetRegion(d, config)
+	v3Client, err := config.NetworkingV3Client(region)
 	if err != nil {
 		return fmtp.DiagErrorf("Error creating HuaweiCloud networking v3 client: %s", err)
 	}
 
-	// only name and enterprise_project_id are supported
-	createOpts := groups.CreateOpts{
+	// Only name and enterprise_project_id are supported
+	createOpts := v3Groups.CreateOpts{
 		Name:                d.Get("name").(string),
 		EnterpriseProjectId: GetEnterpriseProjectID(d, config),
 	}
 
 	logp.Printf("[DEBUG] Create HuaweiCloud Security Group: %#v", createOpts)
-	securityGroup, err := groups.Create(client, createOpts)
+	securityGroup, err := v3Groups.Create(v3Client, createOpts)
 	if err != nil {
+		if _, ok := err.(golangsdk.ErrDefault404); ok {
+			err = createSecurityGroupByV1Client(d, config)
```

Comment:
ditto. I would suggest name it to `resourceNetworkingSecGroupCreateV1` as a common Create function and return resourceNetworkingSecGroupReadV1 directly inside.

Reply:
done.